client/http-client/src/client.rs

```diff
@@ -477,7 +476,12 @@ where
 
 impl<S> SubscriptionClientT for HttpClient<S>
 where
-	S: RpcServiceT<Error = Error, Response = MethodResponse> + Send + Sync,
+	S: RpcServiceT<
+			MethodResponse = Result<MiddlewareMethodResponse, Error>,
+			BatchResponse = Result<MiddlewareBatchResponse, Error>,
+			NotificationResponse = Result<MiddlewareNotifResponse, Error>,
+		> + Send
```

Comment:
Nit: Since `S` isn't used in the impl, can we remove most/all of these constraints here?

Reply:
It's needed unfortunately because the trait SubscriptionClientT: ClientT and ClientT which requires the concrete types to satisfy the backend/server impl :(

Reply:
Aah gotcha!